dev/io.openliberty.security.common.jwt/src/io/openliberty/security/common/jwt/jws/JwsSignatureVerifier.java

```diff
@@ -82,15 +95,19 @@ void verifySignAlgOnly(JsonWebSignature signature) throws EncodedSignatureEmptyE
         if (signature.getEncodedSignature().isEmpty()) {
             throw new EncodedSignatureEmptyException();
         }
-        if (!(signatureAlgorithm.equals(algHeader))) {
+        if (signatureAlgorithm != null && !(signatureAlgorithm.equals(algHeader))) {
             throw new SignatureAlgorithmDoesNotMatchHeaderException(signatureAlgorithm, algHeader);
         }
```

Comment:
i wonder if there is still a use case for this, or if we can just remove the use of `signatureAlgorithm`, since before we were basically comparing the alg header value to the alg header value